core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1605,8 +1613,8 @@ droppingWhile p = droppingWhileM (return . p)
 {-# INLINE takeEndBy_ #-}
 takeEndBy_ :: Monad m => (a -> Bool) -> Fold m a b -> Fold m a b
 -- takeEndBy_ predicate = scanMaybe (takingEndBy_ predicate)
-takeEndBy_ predicate (Fold fstep finitial fextract) =
-    Fold step finitial fextract
+takeEndBy_ predicate (Fold fstep finitial fextract ffinal) =
+    Fold step finitial fextract ffinal
```

Comment:
I don't know if I understand this properly. Why don't we finalize this fold on `Done`?

Reply:
Ah, the one who initializes, finalizes.

Reply:
Only if you initialize it more than once you finalize it.

Reply:
On `Done` we need to finalize this fold.